core/modules/sql/SqlConnection.cc

```diff
@@ -126,14 +127,29 @@ SqlConnection::~SqlConnection() {
 
 bool
 SqlConnection::connectToDb(SqlErrorObject& errObj) {
-    assert(_connection.get());
-    if (_connection->connected()) return true;
+    assert(_connection != nullptr);
+    static int lastChecked = 0;
+    time_t currentTime = time(0);
+
+    if (_connection->connected()) {
+        int rc = mysql_ping(_connection->getMySql());
+        if (rc == 0) {
+            if (currentTime < lastChecked+60) return true;
```

Comment:
Is 60 a magic number? What does this logic try to protect from - disconnects due to inactivity timeout (which should be longer than 60 seconds) or random server disconnects (then this logic does not really work)?